### PR TITLE
Surrogate model based optimization solver 

### PR DESCRIPTION
**Pull request by anujad95**
Expected to be completed by the end of this week. It should take care of #736

### CODE REVIEW

**Comment by anujad95** [comment]
This PR consists of the following:

1. rst files with the detailed documentation and tutorials
2. Aspen, json, FOQUS files for implementing the tutorials

It has been completed. Please review the PR at your earliest convenience. 
Thank you.

File: foqus_lib/framework/optimizer/SM_Optimizer.py [hunk 1]
```diff
@@ -0,0 +1,1191 @@
...
+                    solver_status[i1] = str(r.solver.status)
+                    solver_termination_condition[i1] = str(r.solver.termination_condition)
+                    solver_time[i1] = str(r.solver.time)
+                else:
+                    input_optim[i1] = 1e10
+                    output_optim[i1] = 1e10
+                    objvals[i1] = 1e10
+                    solver_status[i1] = 1e10
+                    solver_termination_condition[i1] = 1e10
+                    solver_time[i1] = 1e10
```
Comment (eslickj) [inline foqus_lib/framework/optimizer/SM_Optimizer.py:505]: Maybe see https://docs.python.org/3/library/math.html#math.inf depending on what you want to do here.  I assumed you just want to start optimal values out really big.  Not sure if inf is appropriate or not, so don't put to much stock in this.

Comment (anujad95) [reply foqus_lib/framework/optimizer/SM_Optimizer.py:505]: Thank you John. I will look into this and get back to you.

Comment (anujad95) [reply foqus_lib/framework/optimizer/SM_Optimizer.py]: I have replaced 1e10 with math.inf. It should work.

File: foqus_lib/framework/optimizer/SM_Optimizer.py [hunk 2]
```diff
@@ -0,0 +1,1191 @@
...
+        self.msgQueue.put("Starting Surrogate based Optimization at {0}".format(
+            time.strftime("%a, %d %b %Y %H:%M:%S", time.localtime())))
+
+        self.msgQueue.put("----------------------")
+        self.bestSoFarList = []
+        
+        #Get user options
+        solversource = self.options['Solver Source'].value
+        
+        mathoptsolver = self.options['mathoptsolver'].value
```
[the remark this answers is not in the document]
Comment (anujad95) [reply foqus_lib/framework/optimizer/SM_Optimizer.py]: I have replaced double spacing with single spacing for the solver options. Let me know if this is fine.

**Pull request merged**